core/src/main/java/nl/nn/adapterframework/stream/Message.java

```diff
@@ -877,6 +874,10 @@ public long size() {
 			return (long) context.get(MessageContext.METADATA_SIZE);
```

Comment:
Cannot this check be moved further up, above the check `if (request instance FileInputStream)`? That's a potentially expensive check when the same is already in the `MessageContext`.
Also, any reason why a size once calculated is not cached in the `MessageContext`?